File: core/src/main/java/com/graphhopper/routing/ev/BooleanEncodedValue.java
package com.graphhopper.routing.ev;

/**
 * This interface defines access to an edge property of type boolean. The default value is false.
 */
public interface BooleanEncodedValue extends EncodedValue {
    void setBool(boolean reverse, int edgeId, EdgeBytesAccess edgeAccess, boolean value);

    boolean getBool(boolean reverse, int edgeId, EdgeBytesAccess edgeAccess);
}

File: core/src/main/kotlin/com/graphhopper/routing/ev/BooleanEncodedValue.kt
package com.graphhopper.routing.ev

/**
 * This interface defines access to an edge property of type boolean. The default value is false.
 */
interface BooleanEncodedValue : EncodedValue {
    fun setBool(reverse: Boolean, edgeId: Int, edgeIntAccess: EdgeIntAccess, value: Boolean)

    fun getBool(reverse: Boolean, edgeId: Int, edgeIntAccess: EdgeIntAccess): Boolean
}
